quantum/process_keycode/process_dynamic_macro.c

```diff
@@ -110,10 +110,12 @@ void dynamic_macro_record_key(keyrecord_t *macro_buffer, keyrecord_t **macro_poi
         **macro_pointer = *record;
         *macro_pointer += direction;
     } else {
-        dynamic_macro_record_key_user(direction, record);
+        // TBD: perform some action when the recording is still in progress but has consumed the available macro buffer
+        dprint("dynamic macro: end of buffer. New key record not added to buffer.\n");
     }
 
     dprintf("dynamic macro: slot %d length: %d/%d\n", DYNAMIC_MACRO_CURRENT_SLOT(), DYNAMIC_MACRO_CURRENT_LENGTH(macro_buffer, *macro_pointer), DYNAMIC_MACRO_CURRENT_CAPACITY(macro_buffer, macro2_end));
+    dynamic_macro_record_key_user(direction, record);
```

Comment:
If you do this, you probably need to change the default implementation of `dynamic_macro_record_key_user()` — currently it blinks the backlight and has a 100 ms delay inside, which could be fine as an error indicator, but would be annoying when actually recording a macro.

Maybe it could be useful to add a `bool success` argument to `dynamic_macro_record_key_user()`, so that this function could be used to indicate errors in some way; this should also break compilation of old code due to prototype mismatch.

(BTW, the implementation of `dynamic_macro_led_blink()` is also not ideal — it performs two EEPROM writes, causing some needless wear.)

Reply:
Ah... I saw the `dynamic_macro_led_blink()` and assumed the intended purpose _was_ to blink at every recorded keystroke to give an indication that it was recording. That may have been the intention, but the reality now is that some users may expect to see it only in a buffer full error case. This could be the action I didn't think of in my placeholder `// TBD: perform some action` , maybe 3 fast blinks, etc.

I like your suggestion of passing an additional parameter like `success` (or something more specific like `buffer_full`?)

Reply:
I pushed another commit which adds the `bool success` argument and used it to restore the previous functionality (blink on error) in the default `dynamic_macro_record_key_user()`. There's probably a debate about what the default behavior should be, but at least this method provides flexibility to cover multiple cases.

There appears to be just one keymap in this repo which implements this user function: keyboards/40percentclub/mf68/keymaps/delivrance/keymap.c
Should PR's include fixes for breaking changes for such keymaps?